seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/source/reader/WordReadStrategy.java

```diff
@@ -0,0 +1,296 @@
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.connectors.seatunnel.file.exception.FileConnectorErrorCode;
+import org.apache.seatunnel.connectors.seatunnel.file.exception.FileConnectorException;
+
+import org.apache.poi.xwpf.usermodel.IBodyElement;
+import org.apache.poi.xwpf.usermodel.XWPFDocument;
+import org.apache.poi.xwpf.usermodel.XWPFHyperlink;
+import org.apache.poi.xwpf.usermodel.XWPFHyperlinkRun;
+import org.apache.poi.xwpf.usermodel.XWPFParagraph;
+import org.apache.poi.xwpf.usermodel.XWPFRun;
+import org.apache.poi.xwpf.usermodel.XWPFTable;
+import org.apache.poi.xwpf.usermodel.XWPFTableCell;
+import org.apache.poi.xwpf.usermodel.XWPFTableRow;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+@Slf4j
+public class WordReadStrategy extends AbstractReadStrategy {
+
+    private static final String DEFAULT_TEXT_COLOR = "000000";
+
+    @Override
+    public void read(String path, String tableId, Collector<SeaTunnelRow> output)
+            throws IOException, FileConnectorException {
+        try (FileInputStream fis = new FileInputStream(path);
+                XWPFDocument document = new XWPFDocument(fis)) {
+
+            int elementId = 1;
+
+            processDocumentInOrder(document, output, tableId, elementId);
+        } catch (Exception e) {
+            throw new FileConnectorException(
+                    FileConnectorErrorCode.FILE_READ_FAILED,
+                    "Failed to read Word document: " + path,
+                    e);
+        }
+    }
+
+    private void processDocumentInOrder(
+            XWPFDocument document, Collector<SeaTunnelRow> output, String tableId, int elementId) {
+
+        for (IBodyElement element : document.getBodyElements()) {
+            if (element instanceof XWPFParagraph) {
+                XWPFParagraph paragraph = (XWPFParagraph) element;
+                if (paragraph.getText() != null && !paragraph.getText().trim().isEmpty()) {
+                    SeaTunnelRow row = createParagraphRow(paragraph, elementId++, tableId);
+                    output.collect(row);
+                }
+            } else if (element instanceof XWPFTable) {
+                XWPFTable table = (XWPFTable) element;
+                String tableData = extractTableData(table);
+                if (!tableData.trim().isEmpty()) {
+                    SeaTunnelRow row = createTableRow(tableData, elementId++, tableId);
+                    output.collect(row);
+                }
+            }
+        }
+    }
+
+    @Override
+    public SeaTunnelRowType getSeaTunnelRowTypeInfo(String path) throws FileConnectorException {
+        return new SeaTunnelRowType(
+                new String[] {
```

Comment:
we need those too.
                    "heading_level",
                    "text",
                    "page_number",
                    "position_index",
                    "parent_id",
                    "child_ids"
Just like markdown. We should keep most fields are same.